### PR TITLE
pat: init at 0.15.0

### PR DESCRIPTION
**Pull request by dotemup**
###### Description of changes

New package for Pat, a Winklink client for packet radio email.
Tested basic functionality with packaged telnet connectivity, all working as expected.
Cherry-pick from https://github.com/NixOS/nixpkgs/pull/242764 for maintainers.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotemup** [comment]
Closed issue https://github.com/NixOS/nixpkgs/issues/239533

File: pkgs/applications/radio/pat/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchFromGitHub {
+    owner = "la5nta";
+    repo = "pat";
+    rev = "v${version}";
+    hash = "sha256-ydv7RQ6MJ+ifWr+babdsDRnaS7DSAU+jiFJkQszy/Ro=";
+  };
+
+  vendorHash = "sha256-TMi5l9qzhhtdJKMkKdy7kiEJJ5UPPJLkfholl+dm/78=";
+
```
Comment (Kranzes) [inline pkgs/applications/radio/pat/default.nix:15]:
```suggestion

  ldflags = [ "-s" "-w" ];
```

Comment (dotemup) [reply pkgs/applications/radio/pat/default.nix:15]: Tested working as expected. Updated in latest.

Comment (Kranzes) [reply pkgs/applications/radio/pat/default.nix:15]: waiting for ci...

**Comment by hax404** [comment]
The commits "applications: add pat" and "pat: init at 0.15.0" should be merged.

**Comment by dotemup** [comment]
Updated with combined commits for adding to top-level. I had asked about this before in another PR but no one replied. Seeing as maintainers was requested to be separate I presumed the top-level would be as separate as well. Good to know!

**Pull request merged**